runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,353 @@
+		}
+	}()
+	err = c.dockerClient.CopyToContainer(ctx, response.ID, "/data", readCloser, types.CopyToContainerOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to copy data to container. Provider=%s: %w", models.Docker, err)
+	}
+	return nil
+}
+
+func (c *Client) export(ctx context.Context, config *provider.ScanJobConfig) (io.ReadCloser, error) {
+	objectType, err := config.AssetInfo.Discriminator()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get asset object type. Provider=%s: %w", models.Docker, err)
+	}
+
+	switch objectType {
+	case "ContainerInfo":
+		id, err := getAssetId(config)
+		if err != nil {
+			return nil, err
+		}
+		return c.dockerClient.ContainerExport(ctx, id)
+	case "ContainerImageInfo":
+		name, err := getAssetName(config)
+		if err != nil {
+			return nil, err
+		}
+		// Create an ephemeral container to export asset
+		response, err := c.dockerClient.ContainerCreate(ctx,
+			&container.Config{
+				Image: name,
+			}, nil, nil, nil, "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to create helper container. Provider=%s: %w", models.Docker, err)
+		}
+		defer func() {
+			err = c.dockerClient.ContainerRemove(ctx, response.ID, types.ContainerRemoveOptions{})
+			if err != nil {
+				_ = fmt.Errorf("failed to remove helper container. Provider=%s: %w", models.Docker, err)
+			}
+		}()
+		return c.dockerClient.ContainerExport(ctx, response.ID)
+	default:
+		return nil, fmt.Errorf("get raw contents not implemented for current object type (%s). Provider=%s", models.Docker, objectType)
+	}
+}
+
+func (c *Client) createScanConfigFile(config *provider.ScanJobConfig) error {
+	scanConfigFilePath, err := getScanConfigFileName(config)
+	if err != nil {
+		return err
+	}
+
+	_, err = os.Stat(scanConfigFilePath)
+	if errors.Is(err, os.ErrNotExist) {
+		err = os.WriteFile(scanConfigFilePath, []byte(config.ScannerCLIConfig), 0644)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) createScanContainer(ctx context.Context, config *provider.ScanJobConfig) (string, error) {
+	scanName, err := getScanName(config)
+	if err != nil {
+		return "", err
+	}
+
+	containerId, err := c.getContainerIdFromContainerName(ctx, scanName)
+	if containerId != "" {
+		return containerId, nil
+	}
+
+	pl, err := c.dockerClient.ImagePull(ctx, config.ScannerImage, types.ImagePullOptions{})
+	if err != nil {
+		return "", err
+	}
+	_, _ = io.Copy(io.Discard, pl)
+	_ = pl.Close()
+
+	scanConfigFilePath, err := getScanConfigFileName(config)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := c.dockerClient.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image: config.ScannerImage,
+			Entrypoint: []string{"sh", "-c",
+				fmt.Sprintf(
+					"/app/vmclarity-cli --config /tmp/%s --server %s --asset-scan-id %s",
+					filepath.Base(scanConfigFilePath),
+					config.VMClarityAddress,
+					config.AssetScanID,
+				),
+			},
```

Comment:
This needs to be broken into a string slice command which calls `/app/vmclarity-cli` directly instead of `sh`